apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,408 @@
+          Debug('_connect(): connectedAddress = ' + this.connectedAddress);
+          Debug('_connect(): onsuccess(): resolve');
+          resolve();
+        };
+
+        req.onerror = () => {
+          // No available profiles are connected.
+          Debug('_connect(): onerror(): reject with connection failed');
+          this.connectingAddress = null;
+          reject('connection failed');
+        };
+      });
+    },
+
+    /**
+     * The method will disconnect the connected device currently
+     * with the device's adapter.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @returns {Promise}
+     */
+    _disconnectCurrentConnectedDevice: 
+    function btcm__disconnectCurrentConnectedDevice() {
+      return new Promise((resolve, reject) => {
+        if (!this.connectedAddress) {
+          // There is no connected device now. Resolve directly.
+          Debug('_disconnectCurrentConnectedDevice(): No connected device now');
+          resolve();
+        }
+
+        // Get connected devices from platform side.
+        this._getConnectedDevices().then((connectedDeviceItemsMap) => {
+          // Get connected device currently.
+          var connectedDevice = 
+            connectedDeviceItemsMap[this.connectedAddress].device;
+          Debug('_disconnectCurrentConnectedDevice(): connectedDevice = ' + 
+                JSON.stringify(connectedDevice));
+          if (connectedDevice) {
+            this.disconnect(connectedDevice).then(() => {
+              // Have disconnected current connected device.
+              Debug('_disconnectCurrentConnectedDevice(): ' + 
+                    'Have disconnected current connected device.');
+              resolve();
+            }, (reason) => {
+              Debug('_disconnectCurrentConnectedDevice(): ' + 
+                    'Cannot disconnect current connected device, ' + 
+                    'reject with reason');
+              reject(reason);
+            });
+          } else {
+            // There is no connected device now.
+            Debug('_disconnectCurrentConnectedDevice(): ' + 
+                  'No connected device in map now');
+            resolve();
+          }
+        }, (reason) => {
+          Debug('_disconnectCurrentConnectedDevice(): ' + 
+                'Cannot get connected device, reject with reason');
+          reject(reason);
+        });
+      });
+    },
+
+    /**
+     * The method will disconnect the input device with the device's adapter.
+     *
+     * @access public
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothDevice} device
+     * @returns {Promise}
+     */
+    disconnect: function btcm_disconnect(device) {
+      return new Promise((resolve, reject) => {
+        if (!this._defaultAdapter) {
+          reject('default adapter is not existed!!');
+        }
+
+        var req = this._defaultAdapter.disconnect(device);
```

Comment:
Now a DOM request is actually a promise. We can use `this._defaultAdapter.disconnect(device).then`.